### DISCUSSION TITLE
Search on FileField.name

### DISCUSSION THREAD

**Discussion by 3ng7n33r** [carltongibson/django-filter#1678] [Q&A] [answered]
Picking up where [this conversation](https://github.com/carltongibson/django-filter/discussions/1478) left off, would it be possible to add a filter for FileFields or maybe find a convenient workaround to add to the docs? I have tinkered with custom filters but nothing seems to work satisfyingly.

**Comment by carltongibson** [comment, marked as answer]
Hi @3ng7n33r 

So the example from has #1478 this: 

```
class QuestionnaireFilter(FilterSet):
    class Meta:
        model = Questionnaire
        fields = ['file']
        filter_overrides = {
            models.FileField: {
                'filter_class': CharFilter,
                'extra': lambda f: {'lookup_expr': 'exact'},
            },
        }
```

Would adding that to the [filter_overrides doc](https://django-filter.readthedocs.io/en/stable/ref/filterset.html#customise-filter-generation-with-filter-overrides) solve your problem? 

(The bottom line is I'm still not convinced filtering on `FileField` is particularly well defined, so would much rather folks define what they need for their own projects than add something not great to the package itself.)

**Reply by 3ng7n33r** [reply]
Hey @carltongibson ,

Yes, that would totally work. I thought it was confusing that there was just no mention of FileFields in the filter docs even though they are Django native. This explanation + solution was exactly what I was looking for. If it is ok for you, I could draft a PR to update the docs to include this example.

**Reply by carltongibson** [reply]
That would be lovely. Thanks. (I'd extend the existing example. I don't know if you want some additional text, but happy to look at what you come up with.)

**Reply by 3ng7n33r** [reply]
Kept it simple. Just enough so "FileField" and "FileFilter" will come up if someone searches for them.
https://github.com/carltongibson/django-filter/pull/1680
